### PR TITLE
Add source information field in ops update and final Report in DREF event detail form

### PR DESCRIPTION
**Pull request by utsukta1**
## Addresses:
- https://github.com/IFRCGo/go-web-app/issues/526

## Changes
- Added source information field in Event Detail tab of Operational Update and Final report(DREF) 

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/DrefFinalReportForm/EventDetail/index.tsx [hunk 1]
```diff
@@ -14,6 +15,11 @@ import Link from '#components/Link';
 import useTranslation from '#hooks/useTranslation';
 import MultiImageWithCaptionInput from '#components/domain/MultiImageWithCaptionInput';
 
+import SourceInformationInput from '#views/DrefApplicationForm/EventDetail/SourceInformationInput';
+import NonFieldError from '#components/NonFieldError';
+import Button from '#components/Button';
+import { useCallback } from 'react';
+import { randomString } from '@togglecorp/fujs';
```
Comment (shreeyash07) [inline src/views/DrefFinalReportForm/EventDetail/index.tsx]: shift this to top

File: src/views/DrefFinalReportForm/EventDetail/index.tsx [hunk 2]
```diff
@@ -246,6 +274,35 @@ function EventDetail(props: Props) {
...
+                            key={source.client_id}
+                            index={index}
+                            value={source}
+                            onChange={onSourceInformationChange}
+                            onRemove={onSourceInformationRemove}
+                            error={getErrorObject(error?.source_information)}
+                            disabled={disabled}
+                        />
+                    ))}
+                    <div className={styles.actions}>
```
Comment (shreeyash07) [inline src/views/DrefFinalReportForm/EventDetail/index.tsx]: Do we need this div? we can pass classname in button.

File: src/views/DrefOperationalUpdateForm/EventDetail/index.tsx [hunk 3]
```diff
@@ -15,6 +16,11 @@ import Link from '#components/Link';
 import useTranslation from '#hooks/useTranslation';
 import MultiImageWithCaptionInput from '#components/domain/MultiImageWithCaptionInput';
 
+import NonFieldError from '#components/NonFieldError';
+import Button from '#components/Button';
+import SourceInformationInput from '#views/DrefApplicationForm/EventDetail/SourceInformationInput';
+import { useCallback } from 'react';
+import { randomString } from '@togglecorp/fujs';
```
Comment (shreeyash07) [inline src/views/DrefOperationalUpdateForm/EventDetail/index.tsx]: Shit this to top

File: src/views/DrefOperationalUpdateForm/EventDetail/index.tsx [hunk 4]
```diff
@@ -357,6 +386,35 @@ function EventDetail(props: Props) {
...
+                                key={source.client_id}
+                                index={index}
+                                value={source}
+                                onChange={onSourceInformationChange}
+                                onRemove={onSourceInformationRemove}
+                                error={getErrorObject(error?.source_information)}
+                                disabled={disabled}
+                            />
+                        ))}
+                        <div className={styles.actions}>
```
Comment (shreeyash07) [inline src/views/DrefOperationalUpdateForm/EventDetail/index.tsx]: Pass classname to button

**Review by shreeyash07 — CHANGES_REQUESTED** [review]

**Review by shreeyash07 — APPROVED** [review]

**Review by barshathakuri — APPROVED** [review]

**Pull request merged**